src/cli/commands/__tests__/lint.test.ts

```diff
@@ -1,448 +1,172 @@
-          );
-        });
-    });
-
-    describe('when single ruleset option provided', () => {
-      test
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', 'non-existent-path'])
-        .exit(2)
-        .it('outputs "does not exist" error');
+describe('lint', () => {
+  let errorSpy: jest.SpyInstance;
+
+  const results: IRuleResult[] = [
+    {
+      code: 'parser',
+      message: 'Duplicate key: foo',
+      path: ['foo'],
+      range: {
+        end: {
+          character: 17,
+          line: 0,
+        },
+        start: {
+          character: 12,
+          line: 0,
+        },
+      },
+      severity: DiagnosticSeverity.Error,
+    },
+  ];
+
+  beforeEach(() => {
+    (lint as jest.Mock).mockClear();
+    (lint as jest.Mock).mockResolvedValueOnce(results);
+
+    (formatOutput as jest.Mock).mockClear();
+    (formatOutput as jest.Mock).mockReturnValueOnce('<formatted output>');
+
+    (writeOutput as jest.Mock).mockClear();
+    (writeOutput as jest.Mock).mockResolvedValueOnce(undefined);
+
+    errorSpy = jest.spyOn(console, 'error');
+  });
 
-      test
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', invalidRulesetPath])
-        .exit(2)
-        .it('outputs "invalid ruleset" error', ctx => {
-          expect(ctx.stdout).toContain(`/rules/rule-without-given-nor-them should have required property 'given'`);
-          expect(ctx.stdout).toContain(`/rules/rule-without-given-nor-them should have required property 'then'`);
-          expect(ctx.stdout).toContain(`/rules/rule-with-invalid-enum/severity should be number`);
-          expect(ctx.stdout).toContain(
-            `/rules/rule-with-invalid-enum/severity should be equal to one of the allowed values`,
-          );
-        });
+  afterEach(() => {
+    errorSpy.mockRestore();
+  });
 
-      test
-        .stdout()
-        .command(['lint', validCustomOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs no issues', ctx => {
-          expect(ctx.stdout).toContain('No errors or warnings found!');
-        });
+  it('shows help when no document argument is passed', async () => {
+    const output = await run('lint');
+    expect(output).toContain('document  Location of a JSON/YAML document');
+  });
 
-      test
-        .stdout()
-        .command(['lint', validOas3SpecPath, '-r', validRulesetPath])
-        .it('outputs warnings in default format', ctx => {
-          expect(ctx.stdout).toContain('5:10  warning  info-matches-stoplight  Info must contain Stoplight');
-          expect(ctx.stdout).not.toContain('Info object should contain `contact` object');
-          expect(ctx.stdout).toContain('OpenAPI 3.x detected');
-        });
+  it('shows help when invalid arguments are passed', async () => {
+    const output = await run('lint --foo');
+    expect(output).toContain('lint a JSON/YAML document from a file or URL');
```

Comment:
hm.. assuming this indeed is help for invalid arguments. If something relating to `foo` is printed, can we have the expectation with `foo` in it? If not, then it OK I guess :)

Reply:
Help shows up whenever there is invalid arguments, so here are two issues:

1. the lack of the required `{document}` argument
2. the --foo option is nonsense

Either one of these two things would cause the help menu to display, but it will not complain about --foo.

Reply:
Actually it probably should. This is copied from Prism, and I thought prism was only doing it when the argument was missing, not invalid options too.

- `spectral lint` should show help
- `spectral lint --foo` should complain about `--foo`